builds/e2e/upgrade-scenarios.yaml

```diff
@@ -0,0 +1,229 @@
+    source: 'Azure-IoT-Edge-Core Build Images'
+    branch: 'release/1.1'
+  - pipeline: images_release_1_2
+    source: 'Azure-IoT-Edge-Core Build Images'
+    branch: 'release/1.2'
+
+jobs:
+################################################################################
+  - job: partial_upgrade_ubuntu_1804_msmoby
+################################################################################
+    displayName: Partial Upgrade
+
+    pool:
+      name: $(pool.linux.name)
+      demands:
+      - ImageOverride -equals agent-aziotedge-ubuntu-18.04-msmoby
+
+    variables:
+      os: linux
+      arch: amd64
+      artifactName: iotedged-ubuntu18.04-amd64
+      identityServiceArtifactName: packages_ubuntu-18.04_amd64
+      identityServicePackageFilter: aziot-identity-service_*_amd64.deb
+      minimal: true
+      verbose: false
+
+    strategy:
+      matrix:
+        # Scenarios 1 to 5 test older versions of the bootstrap Edge Agent 
+        # module with the 1.3 Edge Daemon and Runtime Modules. These represent
+        # scenarios where the Edge Daemon and Runtime Modules in the deployment 
+        # are upgraded, but the bootstrap agent in config.toml/config.yaml is not.
+        1.1.0_to_1.3_no_bootstrap_agent:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.1.0"
+          runtimeModulesVersion: "release/1.3"
+        1.1.8_to_1.3_no_bootstrap_agent:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.1.8"
+          runtimeModulesVersion: "release/1.3"
+        1.1.13_to_1.3_no_bootstrap_agent:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.1.13"
+          runtimeModulesVersion: "release/1.3"
+        1.2.0_to_1.3_no_bootstrap_agent:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.2.0"
+          runtimeModulesVersion: "release/1.3"
+        1.2.9_to_1.3_no_bootstrap_agent:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.2.9"
+          runtimeModulesVersion: "release/1.3"
+
+        # Scenarios 6 to 10 test the 1.3 Edge Daemon with older versions of the 
+        # bootstrap Edge Agent, and runtime modules. These represent scenarios where
+        # the Edge Daemon is upgraded, but the config.toml/config.yaml and deployment 
+        # are left using older runtime images.
+        1.2.0_to_1.3_only_edge_daemon:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.2.0"
+          runtimeModulesVersion: "1.2.0"
+        1.2.9_to_1.3_only_edge_daemon:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.2.9"
+          runtimeModulesVersion: "1.2.9"
+        # TODO: Uncomment if/when the .NET test framework is updated to allow overriding
+        # the default 1.2 schema version used in the EH desired properties section
+        # of the deployment file. Until then, these scenarios will fail
```

Comment:
So, are we leaving important scenarios untested here? Or are these cases we don't expect will happen in practice anyway?

Reply:
I do think these are important scenarios to test, especially since our RQT info shows a large group of customers currently using 1.1.8 and it seems likely that some of them may upgrade the edge host level services to 1.3 but keep the deployments using 1.1.8.

I'm confident that these scenarios work at this point in time aside from the test framework error mentioned in the TODO. But to your point, by commenting these out we lose the automatic regression testing as 1.3 changes.

I'll take a closer look at the test framework and see if there's a reasonable workaround I can implement.

Reply:
I was able to add an `edgeHubSchemaVersion` to the Context class and context.json. If it's missing, it'll default to 1.2. I also updated the logic in the YAML to set the schemaVersion to 1.1 when EH 1.1 is used. Now, the 3 tests that were previously failing are passing and I've uncommented them.

I had to move Context.rs from the `Microsoft.Azure.Devices.Edge.Test` namespace/package to `Microsoft.Azure.Devices.Edge.Test.Common` to avoid the circular dependency that was introduced when I updated `HubModuleConfigBuilder` to use the `Context` class, but I think that's fine because the common package seems like a better fit for it anyway.

Reply:
The Common namespace is a set of tools that make writing tests easier, e.g., a function that deploys a new config, or a function that looks for certain messages arriving in IoT Hub. The values of the Context class, on the other hand, are very specific to the set of tests in the Test namespace. That's why I kept that class out of the Common namespace.

However, putting it in Common probably doesn't hurt anything. We can always adjust later if needed.